OneSignal/onesignal-expo-plugin
# Provide ability to specify both large and small icons.

# Description
## One Line Summary
Allow the Android large and small icons to be specified in the `app.json` file.

## Details
By providing `largeIcons` and `smallIcons` options for the `onesignal-expo-plugin` plugin, an app can add custom icons for Android, described in detail [here](https://documentation.onesignal.com/docs/customize-notification-icons).  These options make it easier for the app to stay in a managed workflow as no platform-specific actions are now required to provide custom icons on Android.

### Small Icons
Small icons specified in the plugin configuration will be copied from the local path provided and distributed into the various [Android resource folders](https://documentation.onesignal.com/docs/customize-notification-icons#default-icon-name-density-size), scaling the original image appropriately.  The plugin will also honor the `app.json` `notification.icon` property as an additional small icon.

Example:  `"smallIcons": ["./assets/ic_stat_onesignal_default.png", "./assets/other_small_icon.png"]`

### Large Icons
Large icons specified in the plugin configuration will be copied from the local path provided and distributed to the XXXHDPI Android resource folder. The input image is recommended to be 256x256.

Example:  `"largeIcons": ["./assets/ic_onesignal_large_icon_default.png", "./assets/other_large_icon.png"]`

### Motivation
Fixes #50 

### Scope
This is an enhancement that makes changes to the `expo prebuild` process only, no runtime changes are expected.  The additional parameters will copy files from the local source (typically the `assets` folder) into one or more `android/app/src/main/res/` subdirectories.


# Testing

## Manual testing
Various scenarios revolving around specifying small/large icons via `notification.icon`, `largeIcons`, and `smallIcons` were performed.  The resulting files were reviewed to ensure correct placement and dimensions.  The resulting build was run on an Android emulator and push notifications sent to ensure images are correctly displayed.


# Checklist
## Overview
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR

## Testing

## Final pass
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.
